transformer_engine/pytorch/float8_tensor.py

```diff
@@ -541,12 +537,11 @@ def to_dtype(self, dtype: torch.dtype) -> Float8Tensor:
         )
 
     def _reset_caches(self) -> None:
-        """Reset cached values
-
+        """
+        Set transpose cache as invalid.
         Should be called after any in-place operation.
-
         """
-        self._transpose = None
+        self._transpose_invalid = True
```

Comment:
If the cache is valid by default, then we should only invalidate if the buffer is empty:
```suggestion
        if self._transpose is not None:
            self._transpose_invalid = True
```

Reply:
See above comment. I propose an invalid cache to begin with.